### ISSUE TITLE
Project Test Case 11, Incorrect statistics & timestamps

### ISSUE THREAD

**Issue by xinnyyy** [nus-cs2030/2324-s2#529]
![image](https://github.com/nus-cs2030/2324-s2/assets/149747847/52e491af-23f4-447b-b413-12553883d41e)
currently facing this issue that the number of customers served is incorrectly accumulated as well as the waiting time. the timestamp in which customer 11 is done serving also seems to be incorrect. i only called serviceTime.get() once in serve event. i cant seem to figure out why since it passed for the previous test cases

**Comment by samuelneo** [comment]
you seem to be missing a Wait -> Serve transition for some customers (15, 16, 17) as well as customer 14 being lost. you might want to check your SCC logic for that

**Comment by xman666888** [comment]
my method: Wait -> Serve build a transition Event between them, u need to find the smallest finished time in the serverslist, and storage it into the Transition Event, if the customer is the first in the queue, it can change from the Transition to Serve, else find the smallest finished time in the serverslist, and change to Transition.

**Comment by Jeremythong2002** [comment]
As codecrunch does not flag out you missing the Wait events, it seems like the issue is that the Wait does not result in a Serve Event. 

Double check that those waiting in the self-checkout counters are eventually served after waiting. The logic for waiting should be the same except that if your implementation of the queue of customers for self-checkout counters is to copy the list each time, they should all be the minimum time.

**Comment by nicthepurplejelly** [comment]
Perhaps you could consider checking the updating of availability of your self check out counters. Addtionally, you could also check whether your logic on checking which counter has the lowest available time to serve the next customer is right?

**Comment by yishao17** [comment]
Hi I think you are missing a wait -> hold -> serve or wait -> serve event. I used the hold method. What I did from hold, it can either generate another hold or serve. the logic is 
1. check whether server given is SCC or not 
2. if it is SCC, need to check for earliest time avail of all SCCs then move the timestamp there
3. if it is not SCC, jus need to update the timestamp to the human server timeavail 
4. for hold -> serve, same as above (need see if its SCC or not) 
hope this helps :))
